### ISSUE TITLE
Upgrade to 1.6.2 fails with unknown field

### ISSUE THREAD

**Issue by davidkarlsen** [zalando/postgres-operator#1440]
Please, answer some short questions which should help us to understand your problem / question better?

- **Which image of the operator are you using?** e.g. registry.opensource.zalan.do/acid/postgres-operator:v1.6.2
- Where do you run it: - metal, OpenShift
- **Are you running Postgres Operator in production?** no
- **Type of issue?** Bug report

```
Normal  error  10s (x4 over 63s)  helm-controller  reconciliation failed: Helm upgrade failed: error validating "": error validating data: [ValidationError(OperatorConfiguration.configuration.kubernetes): unknown field "spilo_allow_privilege_escalation" in do.zalan.acid.v1.OperatorConfiguration.configuration.kubernetes, ValidationError(OperatorConfiguration.configuration): unknown field "major_version_upgrade" in do.zalan.acid.v1.OperatorConfiguration.configuration]
```

**Comment by ingvarch** [comment]
I guess you can try to delete this CRD:

```kubectl delete crd operatorconfigurations.acid.zalan.do```

and `helm install` again

or apply v1.6.2 CRD:

```kubectl apply -f https://raw.githubusercontent.com/zalando/postgres-operator/v1.6.2/manifests/operatorconfiguration.crd.yaml```

**Comment by davidkarlsen** [comment]
I believe the error is here: https://github.com/zalando/postgres-operator/blob/master/charts/postgres-operator/values.yaml#L152

**Comment by ingvarch** [comment]
Yes, you are right, but it is a new variable and you are trying to apply new variables, but use old operatorconfigurations CRD. Try to do what I write before.

**Comment by davidkarlsen** [comment]
I had actually done that already, the problem is that the chart-default uses a string and not a boolean.

**Comment by davidkarlsen** [comment]
But yes, when I reported the issue I had not updated the CRD. Updating the CRD will remove the one error, but the last one about `spilo_allow_privilege_escalation` is tied to the chart-default using the wrong type. If I override it using an unquoted value if will pass.

**Comment by ingvarch** [comment]
Lol, in values-crd.yaml correct value.

https://github.com/zalando/postgres-operator/blob/v1.6.2/charts/postgres-operator/values-crd.yaml#L160

**Comment by FxKu** [comment]
values.yaml -> ConfigMap, which requires bools to be quoted

values-crd.yaml -> OperatorConfiguration CRD, which need bools without quotes.

But, I think even when you choose to use values-crd.yaml, it defaults to values.yaml when no value is found. #1224 should fix it.

**Comment by FxKu** [comment]
We've now merged (#1224) the two values.yaml files. So there should be no more such errors regardless if you choose OperatorConfigurarion CRD or ConfigMap as your config target. On new releases you would need to update the CRD to avoid validation errors. Will close this issue now.
